Makefile

```diff
@@ -83,7 +83,7 @@ export ENABLE_ISTIO_CNI ?= false
 # EXTRA_HELM_SETTINGS="--set istio-cni.excludeNamespaces={} --set istio-cni.tag=v0.1-dev-foo"
 
 
-ISTIO_HELM_REPO := https://raw.githubusercontent.com/istio/istio.io/master/static/charts
+ISTIO_HELM_REPO ?= https://raw.githubusercontent.com/istio/istio.io/master/static/charts
```

Comment:
is this only used to resolve dependency during the build process?

Say if there is a change in one or more istio helm chart yaml, is this or the following "isti%.yaml" target going to use the old copy instead?

Also, is there anything else that needs to be updated in the release process?

Reply:
> is this only used to resolve dependency during the build process?

Yes... This is the default location for the "release" istio-cni chart (which the istio & istio-remote helm charts are dependent on).  It's will take the env var value so it's able to be overridden by the caller of `make generate_yaml` or any target depending on `helm-repo-add`.

> Say if there is a change in one or more istio helm chart yaml, is this or the following "isti%.yaml" target going to use the old copy instead?

"isti%.yaml" targets always use the git workspace's version of install/kubernetes/helm/istio or install/kubernetes/helm/istio-remote charts.  The repo URL pointed to by ISTIO_HELM_REPO is only used to resolve the istio-cni dependency in those charts' requirements.yaml. 

> Also, is there anything else that needs to be updated in the release process?

IMO, we should be able to dump multiple versions of the charts to the same helm repo and specific requirements.yaml references can pin versions.  So, for real  public releases, I'd think we'd want to just be able to push charts to full release chart repo URLs since we should be versioning the charts properly for the release.   For pre-release chart repo URLs, I'm thinking pushing to git repo version specific individual/separate helm repos is useful.  

@sdake  comments ?

Reply:
> Yes... This is the default location for the "release" istio-cni chart (which the istio & istio-remote helm charts are dependent on). It's will take the env var value so it's able to be overridden by the caller of make generate_yaml or any target depending on helm-repo-add.

Except that istio-cni "release" is not fixed... Now that it is built every day (per branch), I suppose we can update it to https://gcsweb.istio.io/gcs/istio-prerelease/daily-build/master-latest-daily/charts/ (or https://storage.googleapis.com/istio-prerelease/daily-build/master-latest-daily/charts/)?

>"isti%.yaml" targets always use the git workspace's version of install/kubernetes/helm/istio or install/kubernetes/helm/istio-remote charts. The repo URL pointed to by ISTIO_HELM_REPO is only used to resolve the istio-cni dependency in those charts' requirements.yaml.

Right. So for a release build, the target will be using the previously built cni "release".

My question is whether it will have any side effect in the produced artifact (i.e. any hard reference to the old tarball)? If so, should that be updated to the new cni "release" built by the current release?

Reply:
> I suppose we can update it to https://gcsweb.istio.io/gcs/istio-prerelease/daily-build/master-latest-daily/charts/ (or https://storage.googleapis.com/istio-prerelease/daily-build/master-latest-daily/charts/)?

Right, I think that's best for now.  But, I'm envisioning it becoming a fixed URL and just the istio/istio/install/kubernetes/helm/requirements.yaml reference a specific version number rather than the ">=x.y.z" ref now.

> My question is whether it will have any side effect in the produced artifact (i.e. any hard reference to the old tarball)? If so, should that be updated to the new cni "release" built by the current release?

From istio/istio, the URL should contain a current release that is known good so any breaking istio/istio changes are found.